src/detection/mod.rs

```diff
@@ -34,17 +37,29 @@ impl<'a> StrDetection<'a> {
 
     /// detect lang with whatlang
     /// if no language is detected, return Language::Other
-    fn detect_lang(text: &str) -> Language {
+    fn detect_lang(text: &str, script: Option<Script>, allow_list : &Option<HashMap<Script,Vec<Language>>>) -> Language {
+        if let Some(script) = script {
+            if let Some(allow_list) = allow_list {
+                if let Some(allow_list) = allow_list.get(&script){
+                    if !allow_list.is_empty() {
+                        let allow_list = allow_list.iter().map(|lang| (*lang).into()).collect();
+                        let detector = Detector::with_allowlist(allow_list);
+                        return detector.detect_lang(text).map(Language::from).unwrap_or_default()
+                    }
+                }
+            }
+        } 
         whatlang::detect_lang(text).map(Language::from).unwrap_or_default()
     }
 }
 
-pub trait Detect {
-    fn detect(&self) -> StrDetection;
+pub trait Detect<'a> {
```

Comment:
I prefer to keep the lifetime name consistency over the whole code, it would be easier to debug in the future.
```suggestion
pub trait Detect<'al> {
```